lib/util/include/hse_util/hse_err.h

```diff
@@ -77,60 +77,44 @@ extern char hse_merr_bug2[];
  *   ------  -----  ----------
  *   63..48   16    signed offset of (_he_merr_file - merr_base) / MERR_ALIGN
  *   47..32   16    line number
- *   31..31    1    reserved bits
- *   30..0    31    positive errno value
+ *   31..16   16    error context
+ *   15..0    16    positive errno value
  */
 #define MERR_FILE_SHIFT (48)
 #define MERR_LINE_SHIFT (32)
-#define MERR_RSVD_SHIFT (31)
+#define MERR_ECTX_SHIFT (16)
 
 #define MERR_FILE_MASK  (0xffff000000000000ul)
 #define MERR_LINE_MASK  (0x0000ffff00000000ul)
-#define MERR_RSVD_MASK  (0x0000000080000000ul)
-#define MERR_ERRNO_MASK (0x000000007ffffffful)
+#define MERR_ECTX_MASK  (0x00000000ffff0000ul)
+#define MERR_ERRNO_MASK (0x000000000000fffful)
 
 typedef s64 merr_t;
 
-#pragma GCC visibility push(default)
-
-static HSE_ALWAYS_INLINE
-uint64_t
+static HSE_ALWAYS_INLINE uint64_t
 merr_to_hse_err(merr_t merr)
 {
     return (uint64_t)merr;
 }
 
-#pragma GCC visibility pop
-
 struct merr_info {
     char buf[MERR_INFO_SZ];
 };
 
 /**
  * merr() - Pack given errno and call-site info into a merr_t
  */
-#define merr(_errnum)   merr_pack((_errnum), _hse_merr_file, __LINE__)
-
-#define merr_once(_errnum)                                              \
-({									\
-    merr_t _err;                                                        \
-                                                                        \
-    if (__builtin_constant_p(_errnum)) {                                \
-        static merr_t _moerr HSE_READ_MOSTLY;                             \
-                                                                        \
-        if (HSE_UNLIKELY(!_moerr))                                          \
-            _moerr = merr_pack((_errnum), _hse_merr_file, __LINE__);    \
-        _err = _moerr;                                                  \
-    } else {								\
-        _err = merr_pack((_errnum), _hse_merr_file, __LINE__);          \
-    }									\
-    _err;                                                               \
-})
+#define merr(_errnum) merr_pack((_errnum), 0, _hse_merr_file, __LINE__)
+
+/**
+ * merrc() - Pack given errno, error context, and call-site info into a merr_t
+ */
+#define merrc(_errnum, _ctx) merr_pack((_errnum), _ctx, _hse_merr_file, __LINE__)
```

Comment:
Parens around `_ctx` in expansion?

Reply:
how about **merrx()** ?  Then if we extended it again we can continue using this interface...